OrbitCore/CrashHandler.cpp

```diff
@@ -10,89 +10,45 @@
-bool DmpFilter(void* context, EXCEPTION_POINTERS* exinfo,
-               MDRawAssertionInfo* assertion) {
-  return true;
+CrashHandler::CrashHandler(const std::wstring& dump_path) {
+  bool not_init = false;
+  if (is_init_.compare_exchange_strong(not_init, true)) {
+    Init(dump_path);
+  }
 }
 
-//-----------------------------------------------------------------------------
-bool DmpCallback(const wchar_t* dump_path, const wchar_t* minidump_id,
-                 void* context, EXCEPTION_POINTERS* exinfo,
-                 MDRawAssertionInfo* assertion, bool succeeded) {
-  PRINT_FUNC;
+void CrashHandler::Init(const std::wstring& dump_path) {
+  // Creates a new CrashpadClient instance that directs crashes to crashpad
+  // handler. Minidump files will be written to dump_path.
 
-  std::string dir = Path::GetDumpPath();
-  std::string msg = "A crash dump was generated in " + dir;
-  MessageBox(NULL,
-             succeeded ? s2ws(msg).c_str() : L"Failed to generate crash dump",
-             L"Orbit Crash Handler: ", MB_ICONEXCLAMATION | MB_OK);
+  TCHAR exe_path[MAX_PATH];
+  GetModuleFileName(NULL, exe_path, MAX_PATH);
+  PathRemoveFileSpec(exe_path);
+  PathAppend(exe_path, L"crashpad_handler.exe");
 
-  return false;
-}
+  base::FilePath dump_file_path(dump_path);
+  base::FilePath handler_file_path(exe_path);
 
-//-----------------------------------------------------------------------------
-bool OnDemandDmpCallback(const wchar_t* dump_path, const wchar_t* minidump_id,
-                         void* context, EXCEPTION_POINTERS* exinfo,
-                         MDRawAssertionInfo* assertion, bool succeeded) {
-  SendDumpInternal(dump_path, minidump_id);
-  return false;
-}
+  std::map<std::string, std::string> annotations;
+  std::vector<std::string> arguments;
 
-//-----------------------------------------------------------------------------
-CrashHandler::CrashHandler() {
-  // Creates a new ExceptionHandler instance to handle writing minidumps.
-  // Before writing a minidump, the optional filter callback will be called.
-  // Its return value determines whether or not Breakpad should write a
-  // minidump.  Minidump files will be written to dump_path, and the optional
-  // callback is called after writing the dump file, as described above.
-  // handler_types specifies the types of handlers that should be installed.
-
-  assert(GHandler == nullptr);
-
-  /*GHandler = new google_breakpad::ExceptionHandler(
-      Path::GetDumpPath().c_str(),
-      DmpFilter,
-      DmpCallback,
-      0,
-      google_breakpad::ExceptionHandler::HANDLER_ALL,
-      MiniDumpNormal,
-      L"",
-      0 );*/
+  crashpad_client_.StartHandler(
+      handler_file_path, /*database=*/dump_file_path,
+      /*metrics_dir=*/dump_file_path, /*url=*/std::string{},
+      /*annotations=*/std::map<std::string, std::string>{},
+      /*arguments=*/std::vector<std::string>{}, /*restartable=*/true,
```

Comment:
You don't need to specify the types here. A single `{}` is fine. That applies to all the three `{}` in here.